src/fairchem/core/trainers/base_trainer.py

```diff
@@ -356,17 +361,128 @@ def load_datasets(self) -> None:
+                    file=elementrefs[target].get("file"),
+                )
+
+        # load or fit normalizers for the dataset.
+        normalizers = self.config["dataset"].get("transforms", {}).get("normalizer", {})
         self.normalizers = {}
-        if normalizer:
-            for target in normalizer:
-                self.normalizers[target] = Normalizer(
-                    mean=normalizer[target].get("mean", 0),
-                    std=normalizer[target].get("stdev", 1),
+        for target in normalizers:
+            if target == "fit" and not normalizers["fit"].get("fitted", False):
+                otf_normalizers = [
+                    {target: None for target in normalizers["fit"]["targets"]}
+                ]
+                # only carry out the fit on master and then broadcast
+                if distutils.is_master():
+                    otf_normalizers = [
+                        fit_normalizers(
+                            targets=normalizers["fit"]["targets"],
+                            element_references=self.elementrefs,
+                            dataset=self.train_dataset,
+                            batch_size=normalizers["fit"].get(
+                                "batch_size", self.config["optim"]["batch_size"]
+                            ),
+                            num_batches=normalizers["fit"].get("num_batches"),
+                            num_workers=self.config["optim"]["num_workers"],
+                            seed=self.config["cmd"]["seed"],
+                        )
+                    ]
+                    # save the normalization for possible subsequent use
+                    if not self.is_debug:
+                        save_checkpoint(
+                            otf_normalizers[0],
+                            self.config["cmd"]["checkpoint_dir"],
+                            "normalizers.pt",
+                        )
+                        logging.info(
+                            f"Normalizers for targets {normalizers['fit']['targets']} have been saved to: {path}"
+                        )
+
+                distutils.broadcast_object_list(otf_normalizers, src=0)
+                self.normalizers.update(otf_normalizers[0])
+                # set config so that normalizers are not refit
+                self.config["dataset"]["transforms"]["normalizer"]["fit"]["fitted"] = (
+                    True
+                )
+            elif target == "file":
+                norms = torch.load(normalizers["file"])
+                self.normalizers.update(norms)
+                logging.info(
+                    f"Loaded normalizers for the following targets: {list(norms.keys())}"
+                )
+            else:
+                self.normalizers[target] = create_normalizer(
+                    file=normalizers[target].get("file"),
+                    mean=normalizers[target].get("mean"),
+                    std=normalizers[target].get("stdev"),
```

Comment:
In this scenario you only need to specify either the file or mean/stdev. Requiring all 3 will throw an error. You should have a fall back value:

```
.get("file", None)
...
```